File: services/api/src/main/java/com/launchly/common/config/EditionConfig.java
package com.launchly.common.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class EditionConfig {
    @Value("${launchly.edition:selfhost}")
    private String edition;

    public String getEdition() {
        return edition;
    }

    public boolean isCloud() {
        return "cloud".equalsIgnoreCase(edition);
    }

    public boolean isSelfHost() {
        return !isCloud();
    }
}

File: services/api/src/common/config/edition.config.ts
import { Injectable } from '@nestjs/common';

@Injectable()
export class EditionConfig {
  private readonly edition = process.env.LAUNCHLY_EDITION || 'selfhost';

  getEdition(): string {
    return this.edition;
  }

  isCloud(): boolean {
    return this.edition === 'cloud';
  }

  isSelfHost(): boolean {
    return !this.isCloud();
  }
}
